cli/src/pcluster/templates/awsbatch_builder.py

```diff
@@ -617,6 +617,7 @@ def _add_manage_docker_images_lambda(self):
                         )
                     ),
                 ],
+                has_vpc_config=self.config.lambda_functions_vpc_config,
```

Comment:
Should `self.config.lambda_functions_vpc_config` be `self.config.deployment_settings.lambda_functions_vpc_config` in every configuration?

Reply:
This is a property (because Python does not have optional chaining/null propagation operator, i.e. like `.?` in JS):

```
    @property
    def lambda_functions_vpc_config(self):
        """Return the vpc config of the PCluster Lambda Functions or None."""
        return self.deployment_settings.lambda_functions_vpc_config if self.deployment_settings else None
```

This is similar to what we do in other cases such as:

```
    @property
    def custom_node_package(self):
        """Return custom node package value or None."""
        return self.dev_settings.node_package if self.dev_settings else None

    @property
    def custom_aws_batch_cli_package(self):
        """Return custom custom aws batch cli package value or None."""
        return self.dev_settings.aws_batch_cli_package if self.dev_settings else None
```